### DISCUSSION TITLE
Retrofit ignores a moshi JsonAdapter

### DISCUSSION THREAD

**Discussion by oleginvoke** [square/retrofit#4032] [Q&A] [answered]
```
Retrofit
            .Builder()
            .baseUrl(ImageSearchApi.BASE_URL)
            .addConverterFactory(
                MoshiConverterFactory.create(
                    Moshi.Builder()
                        .add(SearchFilter::class.java, SearchFilterJsonAdapter())
                        .build()
                ))
            .build()
            .create()
```

```
@Query("tbs")
filter: SearchFilter?
```

Retrofit ignores a moshi JsonAdapter, it's not calling toJson() for the query param

**Comment by JakeWharton** [comment, marked as answer]
Query parameter stringification does not go through the body converters since those can produce non-string content (e.g., protobuf). Instead, it uses a string converter ([documented on the `@Query` docs](https://square.github.io/retrofit/2.x/retrofit/retrofit2/http/Query.html)) to render the value as a string. By default the only one registered is one which calls `toString()`.

If you want to serialize query parameters to JSON using a JSON library, I have written up a sample which does this here: https://github.com/square/retrofit/blob/master/samples/src/main/java/com/example/retrofit/JsonQueryParameters.java
